ts/main.ts

```diff
@@ -58,6 +82,42 @@ function renderAnime(anime: Anime): HTMLDivElement {
   return animeDiv;
 }
 
+// Creates a star button with SVG icon //
+function createStarButton(isFavorite: boolean): HTMLButtonElement {
+  const starButton = document.createElement('button');
+  starButton.classList.add('favorite-button');
+
+  const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+  svg.setAttribute('width', '24');
+  svg.setAttribute('height', '24');
+  svg.setAttribute('viewBox', '0 0 24 24');
+  svg.setAttribute('stroke', 'black');
+  svg.setAttribute('stroke-width', '2');
+  svg.setAttribute('stroke-linecap', 'round');
+  svg.setAttribute('stroke-linejoin', 'round');
+
+  const path = document.createElementNS('http://www.w3.org/2000/svg', 'path');
+  path.setAttribute(
+    'd',
+    'M12 17.27L18.18 21 16.54 13.97 22 9.24 14.81 8.63 12 2 9.19 8.63 2 9.24 7.46 13.97 5.82 21z',
+  );
+  path.setAttribute('fill', isFavorite ? 'yellow' : 'none');
+  svg.appendChild(path);
+  starButton.appendChild(svg);
+
+  return starButton;
+}
```

Comment:
This works, but it may be easier to have used a Font Awesome icon for the star instead.